### DISCUSSION TITLE
Wires created in demo projects

### DISCUSSION THREAD

**Discussion by Sacoder0000** [FPGA-Research/FABulous#254] [Q&A] [answered]
Hello!
In the demo project fabric.csv, the wire from N4BEG to N4END is defined as a 4-bit wire with 4 Y-offset, but the actual generated wire is not(efpga_top.v schema is shown below). Why is the 16-bit wire generated and connected to the adjacent LUT tile? The visualization is generated by vivado.
![temp2](https://github.com/user-attachments/assets/5f063ca1-d362-4cdd-9733-63f418c5cf93)
![temp3](https://github.com/user-attachments/assets/0d599753-7bba-4b7f-99f6-aeed2c47ed89)

**Comment by Sacoder0000** [comment, marked as answer]
Problem solved:
The switch matrices see only the wires amount of wires, regardless of the span. However, the tile-to-tile interfaces include all nested wires concatenated together to a wide vector. FABulous connects the wires LSBs to the switch matrix inputs and the switch matrix outputs are connected to the wires MSBs. Inside the tile, the wide vector is shifted by wires before routing it to the next tile.

**Reply by KelvinChung2000** [reply]
Yes. The details of the wire cascade are also mentioned in the documentation.

**Reply by Sacoder0000** [reply]
Well, I still don't understand what's the point of MID wire, since tile can automatically handle wires that span more than one tile. Is MID specifically for wires with span of 2?

**Reply by IAmMarcelJung** [reply]
The `MID` wires are explained in the documentation, in chapter [fabric definition](https://fabulous.readthedocs.io/en/latest/fabric_definition.html). Different to other wires that are just routed through the tiles _**without**_ connecting to the switch matrix, a `MID` wire **_is_** connected to the switch matrix of the tile it passes.

More information can also be found in #140 (discussions were not available at that point) and #167.

Another thing you might find useful to see how the wires work is the GUI [FABulator](https://github.com/FPGA-Research-Manchester/FABulator).

If you need more information, feel free to ask, but consider opening new discussions for new topics :) 

Also, if you think certain things are not explained well in the documentation, please tell us, so we can improve it (or do it yourself, if you have the time).

**Reply by Sacoder0000** [reply]
Thank you so much! love you~~
